services/src/main/java/org/keycloak/services/resources/admin/RealmAdminResource.java

```diff
@@ -440,8 +440,10 @@ public Response updateRealm(final RealmRepresentation rep) {
             adminEvent.operation(OperationType.UPDATE).representation(StripSecretsUtils.strip(rep)).success();
             
             if (rep.isDuplicateEmailsAllowed() != null && rep.isDuplicateEmailsAllowed() != wasDuplicateEmailsAllowed) {
-                UserCache cache = session.getProvider(UserCache.class);
-                if (cache != null) cache.clear();
+                LegacySessionSupportProvider legacySessionSupportProvider = session.getProvider(LegacySessionSupportProvider.class);
+                if (legacySessionSupportProvider != null) {
+                    legacySessionSupportProvider.clearUserCache();
+                }
```

Comment:
Can you rather use `session.invalidate(InvalidableObjectType.REALM, realm.getId());`?

This would be caught by legacy datastore provider and clear user cache.

Reply:
Updated: 1d94a453bd5c6018c70af2df42e6a37fcdd05bad
Will also put that in the migration docs.